chromadb/db/mixins/sysdb.py

```diff
@@ -426,6 +428,7 @@ def get_collections(
                 .where(databases_t.name == ParameterValue(database))
                 .where(databases_t.tenant_id == ParameterValue(tenant))
             )
+        # cant set limit and offset here because this is metadata and we havent reduced yet
```

Comment:
@jeffchuber, can borrow some of record query code to arrive at this:

```py
        with self.tx() as cur:
            # sql, params = get_sql(q, self.parameter_format())
            # rows = cur.execute(sql, params).fetchall()
            # by_collection = groupby(rows, lambda r: cast(object, r[0]))
            limit = limit or 2**63 - 1
            offset = offset or 0
            self._counter = 0
            collections = list(itertools.islice(self._collections(cur,q),
                                                offset,
                                                min(offset + limit, sys.maxsize)))
            print(f"Number of collections: {self._counter}")
            print(collections)
            return collections

    def _collections(self, cur: Cursor, q: QueryBuilder) -> Generator[Collection, None, None]:
        sql, params = get_sql(q, self.parameter_format())
        cur.execute(sql, params)

        cur_iterator = iter(cur.fetchone, None)
        group_iterator = groupby(cur_iterator, lambda r: r[0])
        for collection_id, group in group_iterator:
            yield self._reduce_collection(collection_id,list(group))
    def _reduce_collection(self,collection_id:str, rows: Sequence[Tuple[Any, ...]])->Sequence[Collection]:
        for row in rows:
            id = self.uuid_from_db(str(collection_id))
            # rows = list(collection_rows)
            name = str(row[1])
            topic = str(row[2])
            dimension = int(row[3]) if row[3] else None
            metadata = self._metadata_from_rows(rows)
            self._counter +=1
            return Collection(
                id=cast(UUID, id),
                topic=topic,
                name=name,
                metadata=metadata,
                dimension=dimension,
                tenant=str(row[5]),
                database=str(row[4]),
```

The assumption is that some users may have 100 or 1000s of collections. The code above is a bit more efficient.